### PR TITLE
Wait for compilation to be available during test setup

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Review by genlu — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
have you checked if this makes thsi pass for you? :)

**Comment by sharwell** [comment]
Not yet, but I will run some iterations locally and see if I can reproduce the bug.

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkaroundsInProcess.cs [hunk 1]
```diff
@@ -17,6 +18,17 @@ namespace Roslyn.VisualStudio.IntegrationTests.InProcess
     [TestService]
     internal partial class WorkaroundsInProcess
     {
+        public async Task WaitForCompilationAvailableAsync(CancellationToken cancellationToken)
```
Comment (jasonmalinowski) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkaroundsInProcess.cs:21]: Something to think about is whether this actually ensures the project exists in the first place from the project system, or whether this is just going to no-op. Might be a useful place to add extra logging.

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkaroundsInProcess.cs [hunk 2]
```diff
@@ -17,6 +18,17 @@ namespace Roslyn.VisualStudio.IntegrationTests.InProcess
     [TestService]
     internal partial class WorkaroundsInProcess
```
Comment (jasonmalinowski) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkaroundsInProcess.cs:19]: Random thought: I love there's a "workarounds" file here so we can easily see where the icky bits are and chase them down long term.

**Review by jasonmalinowski — APPROVED** [review]

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkaroundsInProcess.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkaroundsInProcess.cs:21]: We already wait for project system and workspace.

**Comment by sharwell** [comment]
> have you checked if this makes thsi pass for you? :)

Have now checked locally, and it did not seem to improve this test.

**Pull request closed without merging**